src/ILCompiler.Compiler/src/Compiler/TypeExtensions.cs

```diff
@@ -79,7 +79,7 @@ public static bool RequiresInstMethodTableArg(this MethodDesc method)
         /// </summary>
         public static bool AcquiresInstMethodTableFromThis(this MethodDesc method)
         {
-            return method.IsSharedByGenericInstantiations &&
+            return method.IsCanonicalMethod(CanonicalFormKind.Any) &&
```

Comment:
What is the difference between `IsSharedByGenericInstantiations` differ from `IsCanonicalMethod(CanonicalFormKind.Any)`?